client/app/bundles/course/assessment/containers/AssessmentForm.jsx

```diff
@@ -62,14 +67,51 @@ const validate = (values) => {
 
 class AssessmentForm extends React.Component {
   static propTypes = {
+    dispatch: PropTypes.func.isRequired,
     handleSubmit: PropTypes.func.isRequired,
+    start_at: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.instanceOf(Date),
+    ]),
+    end_at: PropTypes.oneOfType([
+      PropTypes.string,
+      PropTypes.instanceOf(Date),
+    ]),
     autograded: PropTypes.bool,
     password_protected: PropTypes.bool,
     submitting: PropTypes.bool,
     // Above are props from redux-form.
+
     onSubmit: PropTypes.func.isRequired,
+    // If the Form is in editing mode, `published` button will be displayed in editting mode.
+    editing: PropTypes.bool,
+    // If allow to switch between autoraded and manually graded mode.
+    modeSwitching: PropTypes.bool,
+    folderAttributes: PropTypes.shape({
+      folderId: PropTypes.number,
+      // See MaterialFormContainer for detail PropTypes.
+      materials: PropTypes.array,
+    }),
+    // Condtions will be displayed if the attributes are present.
+    conditionAttributes: PropTypes.shape({
+      new_condition_urls: PropTypes.array,
+      conditions: PropTypes.array,
+    }),
   };
 
+  onStartAtChange = (_, newStartAt) => {
```

Comment:
If moving start_at results in a shift in dates, I suspect the user will expect that both dates will shift, if both are present. I think we should minimally implement that. If we want to go all the way, I think we bonus_end_at should shift only when shifting end_at produces an invalid date range, and vice versa. Not sure if that is overkill though.